elastic/beats
# [Heartbeat] Limit jobs by type via Env Vars

Fixes #34290 by allowing users to control job limits via environment vars like `SYNTHETICS_LIMIT_{TYPE}`, so, to limit the max number of  browser jobs one would use: `SYNTHETICS_LIMIT_BROWSER=90 ./heartbeat`
  
This is primarily here to allow users to change the default browser limit in the elastic agent, which is impossible to change otherwise, since there's no way to set a good global config via Fleet.

## Checklist


## How to test this PR locally


`mage build && env SYNTHETICS_LIMIT_BROWSER=4 SYNTHETICS_LIMIT_HTTP=200  ./heartbeat -e -E cloud.id=$CLOUD_ID -E cloud.auth=$CLOUD_AUTH`

Should yield the following logs:

```json
{"log.level":"info","@timestamp":"2023-01-18T16:33:44.124-0600","log.origin":{"file.name":"scheduler/scheduler.go","file.line":79},"message":"limiting to 4 concurrent jobs for 'browser' type","service.name":"heartbeat","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-01-18T16:33:44.124-0600","log.origin":{"file.name":"scheduler/scheduler.go","file.line":79},"message":"limiting to 200 concurrent jobs for 'http' type","service.name":"heartbeat","ecs.version":"1.6.0"}
```
